transport/socks5/stream_dialer.go

```diff
@@ -55,78 +86,144 @@ func (c *streamDialer) DialStream(ctx context.Context, remoteAddr string) (trans
 	}()
 
 	// For protocol details, see https://datatracker.ietf.org/doc/html/rfc1928#section-3
+	// Creating a single buffer for method selection, authentication, and connection request
+	// Buffer large enough for method, auth, and connect requests with a domain name address.
+
+	var buffer []byte
+
+	if c.credentials == nil {
+		// Method selection part: VER = 5, NMETHODS = 1, METHODS = 0 (no auth)
+		// +----+----------+----------+
+		// |VER | NMETHODS | METHODS  |
+		// +----+----------+----------+
+		// | 1  |    1     | 1 to 255 |
+		// +----+----------+----------+
+		header := [3 + 3 + 256 + 2]byte{}
+		buffer = append(header[:0], 5, 1, 0)
```

Comment:
Please clarify the comments:
```suggestion
		// Buffer large enough for method and connect requests with a domain name address.
		header := [3 + (3 + 256 + 2)]byte{}
		// Method selection part: VER = 5, NMETHODS = 1, METHODS = 0 (no auth)
		// +----+----------+----------+
		// |VER | NMETHODS | METHODS  |
		// +----+----------+----------+
		// | 1  |    1     | 1 to 255 |
		// +----+----------+----------+
		buffer = append(header[:0], 5, 1, 0)
```

I also think it's a good idea to add parenthesis to clarify the blocks.